client.go

```diff
@@ -463,23 +463,106 @@ func SetItemCollectionMetrics(client dynamodbiface.DynamoDBAPI, itemCollectionMe
 
 // BatchWriteItemWithContext mock response for dynamodb
 func (fd *Client) BatchWriteItemWithContext(ctx aws.Context, input *dynamodb.BatchWriteItemInput, opts ...request.Option) (*dynamodb.BatchWriteItemOutput, error) {
-	if fd.forceFailureErr != nil {
-		return nil, fd.forceFailureErr
+	return fd.BatchWriteItem(input)
+}
+
+// BatchWriteItem mock response for dynamodb
+func (fd *Client) BatchWriteItem(input *dynamodb.BatchWriteItemInput) (*dynamodb.BatchWriteItemOutput, error) {
+	if err := validateBatchWriteItemInput(input); err != nil {
+		return &dynamodb.BatchWriteItemOutput{}, err
```

Comment:
it should return those items as unprocessed?

Reply:
only the request that can be retried(will be success in the future) are returned in the unprocessed, if the input is not valid now will not be valid in the future.